src/net_processing.cpp

```diff
@@ -1373,7 +1375,10 @@ static void RelayAddress(const CAddress& addr, bool fReachable, const CConnman&
     // Use deterministic randomness to send to the same nodes for 24 hours
     // at a time so the m_addr_knowns of the chosen nodes prevent repeats
     uint64_t hashAddr = addr.GetHash();
-    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr << 32).Write((GetTime() + hashAddr) / (24 * 60 * 60));
+
+    std::chrono::seconds current_time = GetTime<std::chrono::seconds>();
+    uint64_t time_seed = current_time / ROTATE_ADDR_RELAY_DEST_INTERVAL;
+    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr).Write(time_seed);
```

Comment:
```suggestion
    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr).Write((current_time + hashAddr) / ROTATE_ADDR_RELAY_DEST_INTERVAL);
```

Reply:
Yeah, I'm not 100% sure that this measure is effective against anything, but it won't hurt to have a less symmetry.
This mehaviour _may_ leak extra information if someone tries to analyze how addresses are flowing, but since we don't have any real evidence, we probably should stick to the existing reasoning.

Had to slightly adjust the code you suggested, because adding uint to chrono type doesn't work right away.